tests/test_fluid_simulation.py

```diff
@@ -113,3 +108,315 @@ def test_fluid_simulation_phases(self, fluent_simulation):
+        3        9.9700e+02           
+        4        9.9700e+02           
+        5        9.9700e+02           
+        6        9.9700e+02           
+      ...               ...        ...
+"""  # noqa: W291, E501
+        assert str(result) == ref
+        result = cfx_simulation.density_on_nodes(
+            cell_ids=cfx_simulation.mesh.element_ids
+        )
+        assert result.index.mesh_index.location == post.locations.nodal
+        ref = """
+  results     RHO (kg*m^-3)           
+  set_ids                 1           
+    phase Water at 25 C (2) Copper (3)
+ node_ids                             
+     3149        9.9700e+02           
+     4143        9.9700e+02           
+     3140        9.9700e+02           
+     3158        9.9700e+02           
+     3154        9.9700e+02           
+     4146        9.9700e+02           
+      ...               ...        ...
+"""  # noqa: W291, E501
+        assert str(result) == ref
+        result = cfx_simulation.density_on_nodes(face_ids=cfx_simulation.mesh.face_ids)
+        assert result.index.mesh_index.location == post.locations.nodal
+        ref = """
+  results     RHO (kg*m^-3)           
+  set_ids                 1           
+    phase Water at 25 C (2) Copper (3)
+ node_ids                             
+"""  # noqa: W291, E501
+        assert str(result) == ref
+
+    def test_results_cfx_cross_locations_on_faces(self, cfx_simulation):
+        #         result = cfx_simulation.density_on_faces(
+        #             cell_ids=cfx_simulation.mesh.element_ids
+        #         )
+        #         assert result.index.mesh_index.location == post.locations.faces
+        #         assert result._fc[0].location == post.locations.faces
+        #         ref = """
+        #   results     RHO (kg*m^-3)
+        #   set_ids                 1
+        #     phase Water at 25 C (2) Copper (3)
+        #  face_ids
+        #         1        9.9700e+02
+        #         2        9.9700e+02
+        #         3        9.9700e+02
+        #         4        9.9700e+02
+        #         5        9.9700e+02
+        #         6        9.9700e+02
+        #       ...               ...        ...
+        # """  # noqa: W291, E501
+        #         assert str(result) == ref
+        #         result.plot()
+        result = cfx_simulation.density_on_faces(face_ids=cfx_simulation.mesh.face_ids)
+        assert result.index.mesh_index.location == post.locations.faces
+        ref = """
+  results     RHO (kg*m^-3)           
+  set_ids                 1           
+    phase Water at 25 C (2) Copper (3)
+ face_ids                             
+"""  # noqa: W291, E501
+        assert str(result) == ref
+        with pytest.raises(
+            ValueError, match="Cannot plot a Dataframe with an empty mesh index."
+        ):
+            result.plot()
+
+    def test_results_cfx_cross_locations_on_cells(self, cfx_simulation):
+        result = cfx_simulation.density_on_cells(
```

Comment:
Does it make sense to add an scenario of density_on_cells but with nodal scoping?

Reply:
we do not plan to support nodal_scoping to cell_scoping for now, due to the question of inclusive or not, this may be added in a next version.
Sort of like for faces from cell_ids.